apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/index.js

```diff
@@ -45,17 +45,25 @@ if ( window.starterPageTemplatesConfig ) {
 
 const { siteInformation = {}, templates = [] } = window.starterPageTemplatesConfig;
 const editorDispatcher = dispatch( 'core/editor' );
+const editorSelector = select( 'core/editor' );
 
 const insertTemplate = template => {
 	// Skip inserting if there's nothing to insert.
 	if ( ! has( template, 'content' ) ) {
 		return;
 	}
 
-	// set title
-	editorDispatcher.editPost( { title: replacePlaceholders( template.title, siteInformation ) } );
+	// Set post title and remember selected template in meta.
+	const currentMeta = editorSelector.getEditedPostAttribute( 'meta' );
+	editorDispatcher.editPost( {
+		title: replacePlaceholders( template.title, siteInformation ),
+		meta: {
+			...currentMeta,
+			starter_page_template: template.slug, // TODO: change for the proper identifier
```

Comment:
@michaeldcain Should we maintain the `hs_custom_meta` meta key here or roll our own?

Reply:
My idea was to use a key specific for this plugin so we don't have any close ties to our internal Headstart tech and we could easily reuse this plugin with templates from other sources. 

I still want to use the value of `hs_custom_meta` (example: `_hs_front_page`) but it will be placed into that generic sounding meta key. 

One more question for more WP fluent friends here. Should I start my meta key with an underscore to make it private? And should it be prefixed with something Full Site Editing related or are we fine with SPT?

Reply:
> Should I start my meta key with an underscore to make it private?

Yes, I think so. It probably doesn't need to be added to the rest api either, unless I'm missing something.

> should it be prefixed with something Full Site Editing related or are we fine with SPT?

Let's go with what you have so far. We can always change it later if needed

Reply:
I've learned meta needs to be enabled in REST so Gutenberg can access it as Gutenberg uses REST behind the scenes to do all the loading/saving.

Reply:
Ah, that may have been what I missed. Do we need Gutenberg to know about it?

Reply:
Without `show_in_rest: true`, the REST endpoint Gutenberg is calling to save the post will ignore the meta key and it won't get saved. 

I just tested that on my local install with `show_in_rest: false` and I'm not getting anything in the database, despite setting the meta `_starter_page_template` properly in the editor. 

<img width="545" alt="Screenshot 2019-06-06 at 16 56 56" src="https://user-images.githubusercontent.com/156676/59043203-1d469d00-887c-11e9-9c56-cc00a431ada9.png">

Reply:
Thanks for your patience with me :)

Reply:
haha 😀I'm happy I dug deeper and learned something new about the REST API design today!